### PR TITLE
fix(ingest/oracle) add database name to oracle urn name

### PR DESCRIPTION
**Pull request by jaegwonseo**
## Summary
change urn name from oracle ingest like [postgresql](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/postgres.py#L104)
```
schema.table -> database.schema.table
```
## Issue
https://github.com/datahub-project/datahub/issues/6977

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by jjoyce0510 — COMMENTED** [review]
Hi there! This PR is somewhat concerning because it will change the URN structure for existing users of the Oracle source.

This means that if users are NOT using stateful ingestion, then they will end up with duplicates entities in DataHub representing the same tables.

I'm going to enlist @mayurinehate and @hsheth2 to evaluate the changes to understand why this was not included as part of the URN in the first place.

If this indeed is the correct approach (e.g. fixing a bug), we'll need to add specific instructions and notes inside of [updating-datahub.md](https://github.com/datahub-project/datahub/find/master) so that existing users of this source can have an easier time upgrading.

Cheers
John

File: metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py [hunk 1]
```diff
@@ -81,6 +81,15 @@ def get_sql_alchemy_url(self):
             url = f"{url}/?service_name={self.service_name}"
         return url
 
+    def get_identifier(self: BasicSQLAlchemyConfig, schema: str, table: str) -> str:
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py:84]: This database name should also be used in generating the Container URN (schemas!)

Also, are we able to start minting Container nodes for Databases?

**Comment by jjoyce0510** [comment]
Ping on this one. Do you mind adding a note for this change in `updating-datahub.md`?

**Comment by jaegwonseo** [comment]
> updating-datahub.md
@jjoyce0510 
i'm sorry for late replying 
if this approach is correct, i'm gonna update updating-datahub.md

**Comment by jjoyce0510** [comment]
Yes i think it looks okay. Cheers!

File: metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py [hunk 2]
```diff
@@ -81,6 +81,15 @@ def get_sql_alchemy_url(self):
...
+        # schema, table is already normalized, we just to normalize database
+        regular = f"{schema}.{table}"
+        if self.database_alias:
+            return f"{self.database_alias.lower()}.{regular}"
+        if self.database:
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py:89]: Question - why do we need to lower the database here? I'd prefer to retain casing where possible, and lower at a global place

**Comment by jjoyce0510** [comment]
(One comment)

File: metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py (hunk 2 above)
Comment (jaegwonseo) [reply metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py:89]: schema, table is already normalized from below
https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py#L100
https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py#L132

so I tried to normalize it inside this function.

but i'think, it's better to normalize at [global place](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py#L551) 

i remove lower

**Review by anshbansal — CHANGES_REQUESTED** [review]
We need a flag to not have this behaviour by default. Orgs might have really large number of datasets. We cannot suddenly do such breaking change as the default.

**Comment by jaegwonseo** [comment]
> We need a flag to not have this behaviour by default. Orgs might have really large number of datasets. We cannot suddenly do such breaking change as the default.

@anshbansal
i add flag **add_database_name_to_urn** to OracleConfig 
plz check it

**Comment by treff7es** [comment]
@jaegwonseo, we had some concerns about this change, and please, can you make it disabled by default?
If this gets merged with default on, then it can potentially break existing Oracle ingestions (if stateful ingestion is enabled, then because of the urn change, it would soft delete the old datasets, which can potentially mean attached aspect as well will be removed like attached glossary terms, lineage links, etc..)

**Comment by jjoyce0510** [comment]
Checking in here @jaegwonseo  - are you able to separate this into a flag?

**Comment by jaegwonseo** [comment]
@jjoyce0510 
I already added flag(add_database_name_to_urn)

@treff7es 
add_database_name_to_urn has been added, and since it is false by default, it will basically work the same as before.

sorry for the late reply(covid 19)

**Review by jjoyce0510 — APPROVED** [review]
@jaegwonseo Thank you for the hard work.

The change looks good to me. Will plan to merge once CI is green!

**Comment by jaegwonseo** [comment]
@jjoyce0510
is this [test fail](https://github.com/datahub-project/datahub/actions/runs/4207749561/jobs/7304236339) related this pr?

**Comment by chriscollins3456** [comment]
> is this [test fail](https://github.com/datahub-project/datahub/actions/runs/4207749561/jobs/7304236339) related this pr?

re-running as it looked like an issue with the process (a setup job was still running) and not an actual failed test

**Pull request merged**